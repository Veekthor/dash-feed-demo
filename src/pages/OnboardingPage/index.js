import { StyledOnboardingPage } from "./index.styled";
import LoginForm from "./modules/LoginForm";

const OnboardingPage = () => {
  return (
    <StyledOnboardingPage>
      <p>
        A Safe Place for Students to Connect with Each Other,
        <br />
        Engage in Transactions, and Succeed During their
        <br /> College Career.
      </p>
      <LoginForm />
    </StyledOnboardingPage>
  );
};

export default OnboardingPage;
