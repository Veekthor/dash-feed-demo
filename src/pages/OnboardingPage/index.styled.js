import styled from "styled-components";

export const StyledOnboardingPage = styled.div`
  background-color: var(--primary);
  height: 100vh;
  display: flex;
  justify-content: space-around;
  align-items: center;

  & > p {
    color: white;
  }
`;
