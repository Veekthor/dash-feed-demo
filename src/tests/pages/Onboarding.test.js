import renderer from "react-test-renderer";
import OnboardingPage from "../../pages/OnboardingPage";
import { MemoryRouter } from "react-router-dom";
import { act, render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AuthContext } from "../../context/authContext";
import { loginResponse } from "../fixtures/loginResponse";

const OnboardingPageWithRouter = ({ path }) => (
  <MemoryRouter initialEntries={[path]}>
    <OnboardingPage />
  </MemoryRouter>
);
describe("Onboarding Tests", () => {
  const localStore = {};
  beforeAll(() => {
    global.Storage.prototype.setItem = jest.fn((key, value) => {
      localStore[key] = value;
    });
    global.Storage.prototype.getItem = jest.fn((key) => localStore[key]);
  });

  afterAll(() => {
    global.Storage.prototype.setItem.mockReset();
    global.Storage.prototype.getItem.mockReset();
    fetchMock.resetMocks();
  });

  describe("/login", () => {
    it("should match snapshots", () => {
      const component = renderer.create(
        <OnboardingPageWithRouter path="/login" />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should log user in", async () => {
      fetchMock.resetMocks();
      fetchMock.mockResponseOnce(JSON.stringify(loginResponse));
      const setUser = jest.fn();
      const {TEST_LOGIN_PWORD, TEST_LOGIN_EMAIL} = process.env;
      act(() => {
        render(
          <AuthContext.Provider value={{ setUser }}>
            <OnboardingPageWithRouter path="/login" />
          </AuthContext.Provider>
        );
      });
      let input = screen.getByLabelText(/university email/i);
      act(() => userEvent.type(input, TEST_LOGIN_EMAIL));
      input = screen.getByLabelText(/Enter a password/i);
      act(() => userEvent.type(input, TEST_LOGIN_PWORD));
      const submitBtn = screen.getByTestId(/loginBtn/i);
      act(() => userEvent.click(submitBtn));
      await waitFor(() => expect(setUser).toHaveBeenCalled());
      expect(setUser).toHaveBeenCalledWith(loginResponse.user);
      expect(fetch).toHaveBeenCalled();
      expect(global.Storage.prototype.setItem).toHaveBeenCalledWith(
        "userDetails",
        JSON.stringify(loginResponse.user)
      );
      expect(global.Storage.prototype.setItem).toHaveBeenCalledWith(
        "token",
        loginResponse.token
      );
    });
  });

  describe("/register", () => {
    it("should match snapshots", () => {
      const component = renderer.create(
        <OnboardingPageWithRouter path="/register" />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should navigate to login page", () => {
      act(() => {
        render(<OnboardingPageWithRouter path="/register" />);
      });
      let input = screen.getByLabelText(/Enter your university email/i);
      act(() => {
        userEvent.type(input, "chuka@cc");
        input = screen.getByText(/Next/i);
        userEvent.click(input);
      });
      const element = screen.getByText("Login");
      expect(element).toBeInTheDocument();
    });
  });
});
