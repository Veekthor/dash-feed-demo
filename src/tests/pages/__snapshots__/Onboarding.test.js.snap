// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Tests /login should match snapshots 1`] = `
<div
  className="sc-aXZVg gvMYYg"
>
  <p>
    A Safe Place for Students to Connect with Each Other,
    <br />
    Engage in Transactions, and Succeed During their
    <br />
     College Career.
  </p>
  <div
    className="sc-gEvEer gFGtKu"
  >
    <div>
      <h2>
        Hey, welcome back!
      </h2>
      <p>
        Continue enjoying amazing discoveries
      </p>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-dcJsrY elahPx"
      >
        <label
          htmlFor="email"
        >
          Enter your university email
          :
        </label>
        <input
          disabled={false}
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="exampl@kumail.com"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="sc-dcJsrY elahPx"
      >
        <label
          htmlFor="password"
        >
          Enter a password
          :
        </label>
        <input
          disabled={false}
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="sc-fqkvVR hsobOz"
        data-testid="loginBtn"
        disabled={false}
        type="submit"
      >
        Login
      </button>
    </form>
    <div
      className="sc-eqUAAy hqsFEm"
    >
      <p>
        Don't have an account ?
         
        <span
          className="text-primary"
        >
          Sign Up
        </span>
      </p>
      <p
        className="text-primary"
      >
        Forgot your password ?
      </p>
      <p>
        By clicking on 'Login' you've agreed to the
        <br />
        <span
          className="text-primary"
        >
          Terms of Use
        </span>
         and
         
        <span
          className="text-primary"
        >
          Privacy Policy.
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Onboarding Tests /register should match snapshots 1`] = `
<div
  className="sc-aXZVg gvMYYg"
>
  <p>
    A Safe Place for Students to Connect with Each Other,
    <br />
    Engage in Transactions, and Succeed During their
    <br />
     College Career.
  </p>
  <div
    className="sc-gEvEer gFGtKu"
  >
    <div>
      <h2>
        Join and discover more
      </h2>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-dcJsrY elahPx"
      >
        <label
          htmlFor="email"
        >
          Enter your university email
          :
        </label>
        <input
          disabled={false}
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="exampl@kumail.com"
          type="email"
          value=""
        />
      </div>
      <div
        className="sc-dcJsrY elahPx"
      >
        <label
          htmlFor="firstName"
        >
          Enter your First Name
          :
        </label>
        <input
          disabled={false}
          id="firstName"
          name="firstName"
          onChange={[Function]}
          placeholder="Ex. John"
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-dcJsrY elahPx"
      >
        <label
          htmlFor="lastName"
        >
          Enter your Last Name
          :
        </label>
        <input
          disabled={false}
          id="lastName"
          name="lastName"
          onChange={[Function]}
          placeholder="Ex. Granit"
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-dcJsrY elahPx"
      >
        <label
          htmlFor="gradYear"
        >
          Enter your Graduation Year
          :
        </label>
        <input
          disabled={false}
          id="gradYear"
          max={2033}
          min={0}
          name="gradYear"
          onChange={[Function]}
          placeholder="2026"
          type="number"
          value=""
        />
      </div>
      <button
        className="sc-fqkvVR hsobOz"
        data-testid="loginBtn"
        disabled={false}
        type="submit"
      >
        Next
      </button>
    </form>
    <div
      className="sc-eqUAAy hqsFEm"
    >
      <p>
        Already have an account ?
         
        <span
          className="text-primary"
        >
          Log In
        </span>
      </p>
      <p>
        By clicking on 'Login' you've agreed to the
        <br />
        <span
          className="text-primary"
        >
          Terms of Use
        </span>
         and
         
        <span
          className="text-primary"
        >
          Privacy Policy.
        </span>
      </p>
    </div>
  </div>
</div>
`;
